devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java

```diff
@@ -147,6 +146,14 @@ public SSID connectToAP(final WifiConfiguration config, Client client) {
             setupRunnables.add(() -> {
                 log.d("Adding network " + configSSID);
                 networkID.set(wifiFacade.addNetwork(config));
+
+                if (networkID.get() == -1) {
```

Comment:
Not understanding your reply, sorry. :)

Seems like one of these things must be true:
1. `addNetwork()` added the network, but reported failure (-1), or...
2. `removeNetwork()` failed (which then caused `addNetwork()` to fail), and we're just using the previous config?

Did your testing suggest which of these it is?